packages/core/content-releases/admin/src/pages/tests/ReleasesPage.test.tsx

```diff
@@ -1,35 +1,117 @@
 import { within, screen } from '@testing-library/react';
-import { render } from '@tests/utils';
+import { render, server } from '@tests/utils';
+import { rest } from 'msw';
 
 import { ReleasesPage } from '../ReleasesPage';
 
+import { mockData } from './mockData';
+
 jest.mock('@strapi/helper-plugin', () => ({
   ...jest.requireActual('@strapi/helper-plugin'),
   // eslint-disable-next-line
   CheckPermissions: ({ children }: { children: JSX.Element}) => <div>{children}</div>
 }));
 
 describe('Releases home page', () => {
-  it('renders correctly the heading content', async () => {
+  it('renders correctly the page with no results (subtitle, no pagination, body content, add release dialog)', async () => {
+    server.use(
+      rest.get('/content-releases', (req, res, ctx) => res(ctx.json(mockData.emptyEntries)))
+    );
     const { user } = render(<ReleasesPage />);
     expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Releases');
-    // if there are 0 releases
-    expect(screen.getByText('No releases')).toBeInTheDocument();
+    const releaseSubtitle = await screen.findByText('No releases');
+    expect(releaseSubtitle).toBeInTheDocument();
```

Comment:
I think you could organize your tests into code blocks to separate what's going on. Mocks go into one code block, calling the function/rendering the component is one block, expect case you are testing can be its own block.

```tsx
// Mock stuff
server.use(
  rest.get('/content-releases', (req, res, ctx) => res(ctx.json(mockData.emptyEntries)))
);

// Do something
const { user } = render(<ReleasesPage />);

// Expect something after doing something
expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Releases');

// Expect something else after doing something
const releaseSubtitle = await screen.findByText('No releases');
expect(releaseSubtitle).toBeInTheDocument();
```